darts/dataprocessing/transformers/base_data_transformer.py

```diff
@@ -2,49 +2,162 @@
+        `Sequence[TimeSeries]`. This class takes care of parallelizing the transformation of multiple `TimeSeries`
+        when possible.
+
         Data transformers requiring to be fit first before calling `transform()` should derive
         from `FittableDataTransformer` instead.
-        Data transformers which are invertible should derive from ´InvertibleDataTransformer´ instead.
+        Data transformers that are invertible should derive from ´InvertibleDataTransformer´ instead.
 
         Parameters
         ----------
-        names
+        name
             The data transformer's name
+        n_jobs
+            The number of jobs to run in parallel. Parallel jobs are created only when a Sequence[TimeSeries] is passed
+            as input to a method, parallelising operations regarding different TimeSeries. Defaults to `1` (sequential).
+            Setting the parameter to `-1` means using all the available processors.
+            Note: for a small amount of data, the parallelisation overhead could end up increasing the total
+            required amount of time.
+        verbose
+            Optionally, whether to print operations progress
         """
         self._name = name
+        self._verbose = verbose
+        self._n_jobs = n_jobs
+
+    def set_verbose(self, value: bool):
+        """
+        Setter for the verbosity status. `True` for enabling the detailed report about scaler's operation progress,
+        `False` for no additional information
+
+        Parameters
+        ----------
+        value
+            New verbosity status
+
+        """
+        raise_if_not(isinstance(value, bool), "Verbosity status must be a boolean.")
+
+        self._verbose = value
+
+    def set_n_jobs(self, value: int):
+        """
+        Sets the number of cores that will be used by the transformer while processing multiple `TimeSeries`. Set to
+        `-1` for using all the available cores.
+        """
+
+        raise_if_not(isinstance(value, int), "n_jobs must be an integer")
+        self._n_jobs = value
 
+    @staticmethod
     @abstractmethod
-    def transform(self, data: T, *args, **kwargs) -> T:
+    def ts_transform(series: TimeSeries) -> TimeSeries:
         """
-        Transform the data.
-        Not implemented in base class and has to be implemented by deriving classes.
+        The function that will be applied to each `TimeSeries` object once the `transform()` function is called. The
+        function must take as first argument a `TimeSeries` object, and return the transformed `TimeSeries` object.
+        If more parameters are added as input in the derived classes, the `_transform_iterator()` should be
+        redefined accordingly, to yield the necessary arguments to this function (See `_transform_iterator()` for
+        further details).
+
+        This method is not implemented in the base class and must be implemented in the deriving classes.
```

Comment:
a small explanation of why this method is static is probably needed (if not for users, at least for the library's developers)